api/filters/visibility_filtering_middleware.go

```diff
@@ -0,0 +1,56 @@
+package filters
+
+import (
+	"context"
+	"errors"
+
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/web"
+)
+
+type visibilityFilteringMiddleware struct {
+	FilteringFunc func(context.Context, string) (*query.Criterion, error)
+}
+
+func (m visibilityFilteringMiddleware) Run(req *web.Request, next web.Handler) (*web.Response, error) {
+	ctx := req.Context()
+	userCtx, ok := web.UserFromContext(ctx)
+	if !ok {
+		return nil, errors.New("No user found")
+	}
+	if userCtx.AuthenticationType != web.Basic {
+		log.C(ctx).Debugf("Authentication is %s, not basic so proceed without visibility filter criteria", userCtx.AuthenticationType)
+		return next.Handle(req)
+	}
+	platform := &types.Platform{}
+	if err := userCtx.Data(platform); err != nil {
+		return nil, err
+	}
+
+	finalQuery, err := m.FilteringFunc(ctx, platform.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceID := req.PathParams["id"]
```

Comment:
The FilteringFunc makes db calls which transfer all filtered resources. Could we instead introduce filtering only for one element instead of creating the finalQuery and then check if the id is in this query? This would save us a bit of db bandwidth :)

Reply:
I will try and see how it looks like